dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/fat/AcmeValidityAndRenewTest.java

```diff
@@ -347,14 +362,302 @@ public void serverRenewOnRestart() throws Exception {
 			assertNotSame("The certificate should have been marked as expired at startup and renewed.",
 					((X509Certificate) startingCertificateChain[0]).getSerialNumber(),
 					((X509Certificate) endingCertificateChain[0]).getSerialNumber());
+
+			assertNotNull("Should log  message that the certificate was renewed",
+					server.waitForStringInLogUsingMark("CWPKI2052I"));
 		} finally {
-			Log.info(this.getClass(), methodName, "TEST 2: Shutdown.");
+			Log.info(this.getClass(), testName.getMethodName(), "TEST 2: Shutdown.");
 
 			/*
 			 * Stop the server.
 			 */
 			server.stopServer("CWPKI2055W"); // we are running with and intentionally short renewBeforeExpiration.
 		}
 
+		/***********************************************************************
+		 * 
+		 * TEST 3: Start with a very short renew period and short cert checker time out,
+		 * causing a renew request at runtime.
+		 * 
+		 **********************************************************************/
+		try {
+			Log.info(this.getClass(), testName.getMethodName(),
+					"TEST 3: Restart with renew time close to validity period and short cert checker.");
+
+			configuration.getAcmeCA().setCertCheckerSchedule((timeBufferToExpire + 1000) + "ms");
+			configuration.getAcmeCA().setCertCheckerSchedule(AcmeConstants.RENEW_CERT_MIN + "ms");
+
+			AcmeFatUtils.configureAcmeCA(server, caContainer, configuration, DOMAINS1);
+
+			server.startServer();
+			AcmeFatUtils.waitForAcmeAppToStart(server);
+			AcmeFatUtils.waitForSslEndpoint(server);
+
+			/*
+			 * Verify that the server got a new certificate at startup
+			 */
+			endingCertificateChain = AcmeFatUtils.assertAndGetServerCertificate(server, caContainer);
+
+			assertNotSame("The certificate should have been marked as expired at startup and renewed.",
```

Comment:
assertNotSame checks whether they are the same exact instance, not that they are equivalent. You may want to assert that the serials are not the same number (not instance)

Reply:
assertThat(value, not(equal(othervalue)))

Reply:
Changed